### DISCUSSION TITLE
Unable to compile ROSCO using any of the forms here: https://rosco.readthedocs.io/en/latest/source/install.html

### DISCUSSION THREAD

**Discussion by PaulKSmith** [NatLabRockies/ROSCO#315] [Q&A] [answered]
I am trying to get 'libDiscon.so' file by compiling myself for DTU 10MW Wind Turbine. However, I am not able to install in my Linux server properly due to following reason. I am wondering shall I downgrade any version of dependencies so that it will be able to install? Any help would be appreciated. 

Here are the commands I am using as mentioned in the website:

>>conda config --add channels conda-forge 
>>conda create -y --name rosco-env python=3.10 
>>conda activate rosco-env 
>>git clone https://github.com/NREL/ROSCO.git
>>cd ROSCO
>>pip install -e .

Building wheels for collected packages: rosco
  Building editable for rosco (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building editable for rosco (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [225 lines of output]
      WARNING setuptools_scm._integration.setuptools pyproject.toml does not contain a tool.setuptools_scm section
      running editable_wheel
      creating /tmp/pip-wheel-mkxld1gs/.tmp-_6keo2fc/rosco.egg-info
      writing /tmp/pip-wheel-mkxld1gs/.tmp-_6keo2fc/rosco.egg-info/PKG-INFO
      writing dependency_links to /tmp/pip-wheel-mkxld1gs/.tmp-_6keo2fc/rosco.egg-info/dependency_links.txt
      writing requirements to /tmp/pip-wheel-mkxld1gs/.tmp-_6keo2fc/rosco.egg-info/requires.txt
      writing top-level names to /tmp/pip-wheel-mkxld1gs/.tmp-_6keo2fc/rosco.egg-info/top_level.txt
      writing manifest file '/tmp/pip-wheel-mkxld1gs/.tmp-_6keo2fc/rosco.egg-info/SOURCES.txt'
      adding license file 'LICENSE'
      writing manifest file '/tmp/pip-wheel-mkxld1gs/.tmp-_6keo2fc/rosco.egg-info/SOURCES.txt'
      creating '/tmp/pip-wheel-mkxld1gs/.tmp-_6keo2fc/rosco-2.9.0.dist-info'
      creating /tmp/pip-wheel-mkxld1gs/.tmp-_6keo2fc/rosco-2.9.0.dist-info/WHEEL
      running build_py
      running build_ext
      -- The Fortran compiler identification is GNU 4.8.5
      -- The C compiler identification is GNU 4.8.5
      -- Detecting Fortran compiler ABI info
      -- Detecting Fortran compiler ABI info - done
      -- Check for working Fortran compiler: /usr/bin/f95 - skipped
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /usr/bin/cc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- CMAKE_Fortran_COMPILER_ID = GNU
      -- Setting system file as: src/SysFiles/SysGnuLinux.f90
      -- Found PkgConfig: /usr/bin/pkg-config (found version "0.27.1")
      -- Checking for module 'libzmq'
      --   No package 'libzmq' found
      -- Configuring done (1.9s)
      -- Generating done (0.0s)
      CMake Warning:
        Manually-specified variables were not used by the project:

          CALL_FROM_SETUP_PY
          Python3_ROOT_DIR

      -- Build files have been written to: /tmp/tmplh10r6m6.build-temp_rosco
      [1/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/Constants.f90-pp.f90
      [2/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/ROSCO_Types.f90-pp.f90
      [3/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/Controllers.f90-pp.f90
      [4/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/ExtControl.f90-pp.f90
      [5/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/ZeroMQInterface.f90-pp.f90
      [6/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/DISCON.F90-pp.f90
      [7/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/Functions.f90-pp.f90
      [8/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/ControllerBlocks.f90-pp.f90
      [9/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/Filters.f90-pp.f90
      [10/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/ROSCO_Helpers.f90-pp.f90
      [11/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/ROSCO_IO.f90-pp.f90
      [12/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/SysFiles/SysGnuLinux.f90-pp.f90
      [13/28] Building Fortran preprocessed CMakeFiles/discon.dir/src/ReadSetParameters.f90-pp.f90
      [14/28] Generating Fortran dyndep file CMakeFiles/discon.dir/Fortran.dd
      [15/28] Building Fortran object CMakeFiles/discon.dir/src/Constants.f90.o
      [16/28] Building Fortran object CMakeFiles/discon.dir/src/ROSCO_Types.f90.o
      FAILED: CMakeFiles/discon.dir/src/ROSCO_Types.f90.o ftnmods/rosco_types.mod
      /usr/bin/f95 -I/home/usr123/ROSCO/rosco/controller/src -ffree-line-length-0 -DIMPLICIT_DLLEXPORT -ffree-line-length-0 -fdefault-real-8 -fdefault-double-8 -cpp -O3 -DNDEBUG -O3 -Jftnmods -fPIC -fpreprocessed -c CMakeFiles/discon.dir/src/ROSCO_Types.f90-pp.f90 -o CMakeFiles/discon.dir/src/ROSCO_Types.f90.o
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:346:107: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmTDX                     ! Platform motion -- Displacement TDX (m)')
                                                                                                                 ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:347:107: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmTDY                     ! Platform motion -- Displacement TDY (m)')
                                                                                                                 ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:348:107: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmTDZ                     ! Platform motion -- Displacement TDZ (m)')
                                                                                                                 ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:349:109: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmRDX                     ! Platform motion -- Displacement RDX (rad)')
                                                                                                                   ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:350:109: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmRDY                     ! Platform motion -- Displacement RDY (rad)')
                                                                                                                   ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:351:109: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmRDZ                     ! Platform motion -- Displacement RDZ (rad)')
                                                                                                                   ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:352:109: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmTVX                     ! Platform motion -- Velocity     TVX (m/s)')
                                                                                                                   ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:353:109: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmTVY                     ! Platform motion -- Velocity     TVY (m/s)')
                                                                                                                   ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:354:109: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmTVZ                     ! Platform motion -- Velocity     TVZ (m/s)')
                                                                                                                   ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:355:111: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmRVX                     ! Platform motion -- Velocity     RVX (rad/s)')
                                                                                                                     ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:356:111: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmRVY                     ! Platform motion -- Velocity     RVY (rad/s)')
                                                                                                                     ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:357:111: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmRVZ                     ! Platform motion -- Velocity     RVZ (rad/s)')
                                                                                                                     ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:358:111: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmTAX                     ! Platform motion -- Acceleration TAX (m/s^2)')
                                                                                                                     ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:359:111: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmTAY                     ! Platform motion -- Acceleration TAY (m/s^2)')
                                                                                                                     ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:360:111: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmTAZ                     ! Platform motion -- Acceleration TAZ (m/s^2)')
                                                                                                                     ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:361:113: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmRAX                     ! Platform motion -- Acceleration RAX (rad/s^2)')
                                                                                                                       ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:362:113: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmRAY                     ! Platform motion -- Acceleration RAY (rad/s^2)')
                                                                                                                       ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:363:113: warning: missing terminating ' character [enabled by default]
           REAL(DbKi)                    :: PtfmRAZ                     ! Platform motion -- Acceleration RAZ (rad/s^2)')
                                                                                                                       ^
      /home/usr123/ROSCO/rosco/controller/src/ROSCO_Types.f90:436.43:

          CHARACTER(:), ALLOCATABLE     :: ErrMsg                      ! a Fortran ve
                                                 1
      Error: Deferred-length character component 'errmsg' at (1) is not yet supported
      ninja: build stopped: subcommand failed.
      Traceback (most recent call last):
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 156, in run
          self._create_wheel_file(bdist_wheel)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 345, in _create_wheel_file
          files, mapping = self._run_build_commands(dist_name, unpacked, lib, tmp)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 268, in _run_build_commands
          self._run_build_subcommands()
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 295, in _run_build_subcommands
          self.run_command(name)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/cmake_build_extension/build_extension.py", line 113, in run
          self.build_extension(ext)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/cmake_build_extension/build_extension.py", line 230, in build_extension
          subprocess.check_call(build_command)
        File "/home/usr123/anaconda3/envs/rosco-env/lib/python3.10/subprocess.py", line 369, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['cmake', '--build', '/tmp/tmplh10r6m6.build-temp_rosco', '--config', 'Release']' returned non-zero exit status 1.
      /tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py:988: _DebuggingTips: Problem in editable installation.
      !!

              ********************************************************************************
              An error happened while installing `rosco` in editable mode.

              The following steps are recommended to help debug this problem:

              - Try to install the project normally, without using the editable mode.
                Does the error still persist?
                (If it does, try fixing the problem before attempting the editable mode).
              - If you are using binary extensions, make sure you have all OS-level
                dependencies installed (e.g. compilers, toolchains, binary libraries, ...).
              - Try the latest version of setuptools (maybe the error was already fixed).
              - If you (or your project dependencies) are using any setuptools extension
                or customization, make sure they support the editable mode.

              After following the steps above, if the problem still persists and
              you think this is related to how setuptools handles editable installations,
              please submit a reproducible example
              (see https://stackoverflow.com/help/minimal-reproducible-example) to:

                  https://github.com/pypa/setuptools/issues

              See https://setuptools.pypa.io/en/latest/userguide/development_mode.html for details.
              ********************************************************************************

      !!
        cmd_obj.run()

      ==> Configuring:
      $ cmake -S /home/usr123/ROSCO/rosco/controller -B /tmp/tmplh10r6m6.build-temp_rosco -GNinja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX:PATH=/home/usr123/ROSCO/rosco -DCMAKE_MAKE_PROGRAM=/tmp/pip-build-env-x1re7huv/overlay/bin/ninja -DPython3_ROOT_DIR=/home/usr123/anaconda3/envs/rosco-env -DCALL_FROM_SETUP_PY:BOOL=ON -DBUILD_SHARED_LIBS=ON -DCMAKE_Fortran_FLAGS=-ffree-line-length-0

      ==> Building:
      $ cmake --build /tmp/tmplh10r6m6.build-temp_rosco --config Release

      ==> Installing:
      $ cmake --install /tmp/tmplh10r6m6.build-temp_rosco

      Traceback (most recent call last):
        File "/home/usr123/anaconda3/envs/rosco-env/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/usr123/anaconda3/envs/rosco-env/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/usr123/anaconda3/envs/rosco-env/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 273, in build_editable
          return hook(wheel_directory, config_settings, metadata_directory)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 436, in build_editable
          return self._build_with_temp_dir(
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 389, in _build_with_temp_dir
          self.run_setup()
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 56, in <module>
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 156, in run
          self._create_wheel_file(bdist_wheel)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 345, in _create_wheel_file
          files, mapping = self._run_build_commands(dist_name, unpacked, lib, tmp)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 268, in _run_build_commands
          self._run_build_subcommands()
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/command/editable_wheel.py", line 295, in _run_build_subcommands
          self.run_command(name)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/cmake_build_extension/build_extension.py", line 113, in run
          self.build_extension(ext)
        File "/tmp/pip-build-env-x1re7huv/overlay/lib/python3.10/site-packages/cmake_build_extension/build_extension.py", line 230, in build_extension
          subprocess.check_call(build_command)
        File "/home/usr123/anaconda3/envs/rosco-env/lib/python3.10/subprocess.py", line 369, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['cmake', '--build', '/tmp/tmplh10r6m6.build-temp_rosco', '--config', 'Release']' returned non-zero exit status 1.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building editable for rosco
Failed to build rosco
ERROR: Could not build wheels for rosco, which is required to install pyproject.toml-based projects

Here are the dependencies in rosco-env. This is just to confirm wheel is already installed latest version. 
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
ca-certificates           2024.2.2             hbcca054_0    conda-forge
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h807b86a_5    conda-forge
libgomp                   13.2.0               h807b86a_5    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libsqlite                 3.45.1               h2797004_0    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
ncurses                   6.4                  h59595ed_2    conda-forge
openssl                   3.2.1                hd590300_0    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
python                    3.10.13         hd12c33a_1_cpython    conda-forge
readline                  8.2                  h8228510_1    conda-forge
setuptools                69.0.3             pyhd8ed1ab_0    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tzdata                    2024a                h0c530f3_0    conda-forge
wheel                     0.42.0             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge

I also tried installing ROSCO using CMAKE, however, I am facing error there as well. Here it is:
>conda install m2w64-toolchain libpython

Error while loading conda entry point: conda-libmamba-solver (libarchive.so.19: cannot open shared object file: No such file or directory)
CondaValueError: You have chosen a non-default solver backend (libmamba) but it was not recognized. Choose one of: classic

**Comment by dzalkind** [comment, marked as answer]
It looks like the fortran compiler you are using does not like the comments in that `ROSCO_Types.f90` file.

You could try updating the `compilers` package from conda.  GNU 4.8.5 is much older than our usual compilers.

The `conda install m2w64-toolchain libpython` is for native windows, not WSL.

**Reply by PaulKSmith** [reply]
@dzalkind Updating Fortran compiler using `conda install conda-forge::fortran-compiler` helped installing ROSCO. Thanks a lot for your kind response.

**Reply by PaulKSmith** [reply]
I tried to install the same in my Windows PC. I am wondering if I need to install `libzmq`. The build was successful though.

`(rosco-env2) C:\ROSCO\rosco\controller\build>cmake .. -G "MinGW Makefiles"
-- CMAKE_Fortran_COMPILER_ID = GNU
-- Setting system file as: src/SysFiles/SysGnuWin.f90
-- Checking for module 'libzmq'
--   No package 'libzmq' found
-- Configuring done (0.2s)
-- Generating done (0.0s)
-- Build files have been written to: C:/ROSCO/rosco/controller/build`


`(rosco-env2) C:\ROSCO\rosco\controller\build>make install
[100%] Built target discon
Install the project...
-- Install configuration: "RelWithDebInfo"
-- Installing: C:/Program Files (x86)/ROSCO/lib/libdiscon.dll.a
-- Installing: C:/Program Files (x86)/ROSCO/lib/libdiscon.dll`

**Comment by dzalkind** [comment]
Hi Paul,

You will probably need to make another build directory and run `cmake` from there?

Best, Dan

**Reply by PaulKSmith** [reply]
Hi Dan @dzalkind ,

I was having some problem with my PC and took some time to report back.
For my Windows PC I am using these commands. It was successfully installed however, I cannot see `rosco` under `conda list -n rosco-env`. Do I need to use `pip install -e .` as well here to run the python scripts under `examples ` folder? 

I am also wondering how to get the *.dll and *.so files for DTU10MW, provided I have  `AERODYN v15.03.* INPUT FILE`, `DISCON.IN`, `Cp_Ct_Cq_10MW.txt`, `*.yaml` files already. I am going through each example file and not sure which one will help me generate` dll/so `files. If anyone could help it will make things a little easier.

`conda config --add channels conda-forge`
`conda create -y --name rosco-env python=3.10`
`conda activate rosco-env`
`git clone https://github.com/NREL/ROSCO.git`

`conda install m2w64-toolchain libpython`
`conda install cmake make`
`cd ROSCO/rosco/controller`
`mkdir build`
`cd build`
`cmake .. -G "MinGW Makefiles"`

`make install`

`(rosco-env) C:\ROSCO\rosco\controller\build>make install
[  7%] Building Fortran object CMakeFiles/discon.dir/src/Constants.f90.obj
[ 14%] Building Fortran object CMakeFiles/discon.dir/src/ROSCO_Types.f90.obj
[ 21%] Building Fortran object CMakeFiles/discon.dir/src/Filters.f90.obj
[ 28%] Building Fortran object CMakeFiles/discon.dir/src/Functions.f90.obj
[ 35%] Building Fortran object CMakeFiles/discon.dir/src/ControllerBlocks.f90.obj
[ 42%] Building Fortran object CMakeFiles/discon.dir/src/Controllers.f90.obj
[ 50%] Building Fortran object CMakeFiles/discon.dir/src/SysFiles/SysGnuWin.f90.obj
[ 57%] Building Fortran object CMakeFiles/discon.dir/src/ExtControl.f90.obj
[ 64%] Building Fortran object CMakeFiles/discon.dir/src/ROSCO_Helpers.f90.obj
[ 71%] Building Fortran object CMakeFiles/discon.dir/src/ReadSetParameters.f90.obj
[ 78%] Building Fortran object CMakeFiles/discon.dir/src/ROSCO_IO.f90.obj
[ 85%] Building Fortran object CMakeFiles/discon.dir/src/ZeroMQInterface.f90.obj
[ 92%] Building Fortran object CMakeFiles/discon.dir/src/DISCON.F90.obj
[100%] Linking Fortran shared library libdiscon.dll
[100%] Built target discon
Install the project...
-- Install configuration: "RelWithDebInfo"
-- Installing: C:/Program Files (x86)/ROSCO/lib/libdiscon.dll.a
-- Installing: C:/Program Files (x86)/ROSCO/lib/libdiscon.dll`

Regards,
